src/overlays/actors/ovl_En_Heishi3/z_en_heishi3.c

```diff
@@ -27,27 +41,198 @@ const ActorInit En_Heishi3_InitVars = {
+ **/
+void EnHeishi3_CastleHandler(EnHeishi3* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((player->actor.posRot.pos.x < -190.0f) && (player->actor.posRot.pos.x > -380.0f) &&
+        (fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) < 100.0f) &&
+        (player->actor.posRot.pos.z < 1020.0f) && (player->actor.posRot.pos.z > 700.0f) && (isCaught == 0)) {
+        if (this->unk_278 == 1) {
+            if ((player->actor.posRot.pos.x < -290.0f)) {
+                return;
+            }
+        }
+
+        else {
+            if (player->actor.posRot.pos.x > -290.0f) {
+                return;
+            }
+        }
+        isCaught = 1;
+        func_8010B680(globalCtx, 0x702D, &this->actor); // "Hey you! Stop! You, kid, over there!"
+        func_80078884(NA_SE_SY_FOUND);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発見！ ☆☆☆☆☆ \n" VT_RST); // "Discovered!"
+        func_8002DF54(globalCtx, &this->actor, 1);
+        this->actionFunc = EnHeishi3_Caught;
+    }
+}
+
+void EnHeishi3_Caught(EnHeishi3* this, GlobalContext* globalCtx) {
+    f32 frames = SkelAnime_GetFrameCount(&D_06005880.genericHeader);
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005880, 1.0f, 0.0f, (s16)(f32)frames, 0, -10.0f);
+    this->caughtTimer = 20;
+    this->actionFunc = func_80A55BD4;
+    this->actor.speedXZ = 2.5f;
+}
+
+void func_80A55BD4(EnHeishi3* this, GlobalContext* globalCtx) {
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((func_800A56C8(&this->skelAnime, 1.0f) != 0) || (func_800A56C8(&this->skelAnime, 17.0f) != 0)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_KNIGHT_WALK);
+    }
+    if (this->caughtTimer == 0) {
+        this->actionFunc = EnHeishi3_SetupRespawn;
+        this->actor.speedXZ = 0.0f;
+    } else {
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, this->actor.rotTowardsLinkY, 5, 3000, 0);
+    }
+}
+
+void EnHeishi3_SetupRespawn(EnHeishi3* this, GlobalContext* globalCtx) {
+    f32 frames = SkelAnime_GetFrameCount(&D_06005C30.genericHeader);
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005C30, 1.0f, 0.0f, (s16)(f32)frames, 0, -10.0f);
+    this->actionFunc = EnHeishi3_Respawn;
+}
+
+void EnHeishi3_Respawn(EnHeishi3* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && (func_80106BC8(globalCtx) != 0) && (this->respawnFlag == 0)) {
+        gSaveContext.eventChkInf[4] |= 0x4000;
+        globalCtx->nextEntranceIndex = 0x47E; // Hyrule Castle from Guard Capture (outside)
+        globalCtx->sceneLoadFlag = 0x14;
+        this->respawnFlag = 1;
+        globalCtx->fadeTransition = 0x2E;
+        gSaveContext.nextTransition = 0x2E;
+    }
+}
+
+void EnHeishi3_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnHeishi3* this = THIS;
+    ColliderCylinder* cylinderCollider;
```

Comment:
I could either use the temp or just make it a pad. Because stack doesnt match without something there

Reply:
pad imo

Reply:
Yeah pad for sure. The stack padding could be anything, that we can't see, and since using a temp doesn't make the code any better here, there isn't much point.